crates/re_selection_panel/src/visualizer_ui.rs

```diff
@@ -0,0 +1,341 @@
+        let result_override = query_result.overrides.get(component);
+        let raw_override = result_override.and_then(|r| r.try_raw(&query_result.resolver));
+        let non_empty_override = raw_override.as_ref().map_or(false, |r| !r.is_empty());
+
+        let result_store = query_result.results.get(component);
+        let raw_store = result_store.and_then(|r| r.try_raw(&query_result.resolver));
+        let non_empty_store = raw_store.as_ref().map_or(false, |r| !r.is_empty());
+
+        let result_default = query_result.defaults.get(component);
+        let raw_default = result_default.and_then(|r| r.try_raw(&query_result.resolver));
+        let non_empty_default = raw_default.as_ref().map_or(false, |r| !r.is_empty());
+
+        let raw_fallback = match visualizer.fallback_for(&query_ctx, component) {
+            Ok(fallback) => fallback,
+            Err(err) => {
+                re_log::warn_once!("Failed to get fallback for component {component}: {err}");
+                continue; // TODO(andreas): Don't give up on the entire component because of this. Show an error instead.
+            }
+        };
+
+        // Determine where the final value comes from.
+        // Putting this into an enum makes it easier to reason about the next steps.
+        let value_source = match (non_empty_override, non_empty_store, non_empty_default) {
+            (true, _, _) => ValueSource::Override,
+            (false, true, _) => ValueSource::Store,
+            (false, false, true) => ValueSource::Default,
+            (false, false, false) => ValueSource::FallbackOrPlaceholder,
+        };
+
+        #[allow(clippy::unwrap_used)] // We checked earlier that these values are valid!
+        let raw_current_value = match value_source {
+            ValueSource::Override => raw_override.as_ref().unwrap(),
+            ValueSource::Store => raw_store.as_ref().unwrap(),
+            ValueSource::Default => raw_default.as_ref().unwrap(),
+            ValueSource::FallbackOrPlaceholder => &raw_fallback,
+        }
+        .as_ref();
+
+        let Some(override_path) = data_result.individual_override_path() else {
+            // This shouldn't the `DataResult` is valid.
+            if cfg!(debug_assertions) {
+                re_log::error!("No override path for entity: {}", data_result.entity_path);
+            }
+            return;
+        };
+
+        let value_fn = |ui: &mut egui::Ui, _style| {
+            // Edit ui can only handle a single value.
+            let multiline = false;
+            if raw_current_value.len() > 1
+                || !ctx.viewer_ctx.component_ui_registry.try_show_edit_ui(
+                    ctx.viewer_ctx,
+                    ui,
+                    raw_current_value,
+                    override_path,
+                    component,
+                    multiline,
+                )
+            {
+                // TODO(andreas): Unfortunately, display ui wants to do the query itself.
```

Comment:
outdated comment. I fixed that by now!
(also broken comment, guess I went somewhere else halfway of writing this)